diesel_infer_schema/src/pg.rs

```diff
@@ -10,6 +11,14 @@ pub fn determine_column_type(attr: &ColumnInformation) -> Result<ColumnType, Box
         &attr.type_name
     };
 
+    let tpe_is_varchar = tpe.to_lowercase() == "varchar";
+
+    // Postgres doesn't coerce varchar[] to text[] so print out a message to inform
+    // the user.
+    if tpe_is_varchar && is_array {
+        writeln!(&mut stderr(), "The column `{}` is of type `varchar[]`. Array operations are not supported on the `varchar[]` type. You should change your column type to `text[]`.", attr.column_name)?;
```

Comment:
The second sentence is misleading. I'd change it to: `This will cause problems when using Diesel.`